applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py

```diff
@@ -0,0 +1,102 @@
+import KratosMultiphysics as KratosMultiphysics
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return ImposeRigidMovementProcess(Model, settings["Parameters"])
+
+## All the processes python should be derived from "Process"
+class ImposeRigidMovementProcess(KratosMultiphysics.Process):
+    """This class is used in order to impose a rigid body movement in a certain region of the problem
+
+    This class constructs the model parts containing the constrains that enforce the rigid body movement
+    Only the member variables listed below should be accessed directly.
+
+    Public member variables:
+    Model -- the container of the different model parts.
+    settings -- Kratos parameters containing solver settings.
+    """
+
+    def __init__(self, Model, settings ):
+        """ The default constructor of the class
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        Model -- the container of the different model parts.
+        settings -- Kratos parameters containing solver settings.
+        """
+        KratosMultiphysics.Process.__init__(self)
+
+        default_settings = KratosMultiphysics.Parameters("""
+        {
+            "help"                        : "This process uses LinearMasterSlaveConstraint in order to impose an unified movement in the given submodelpart. The process takes the first node from the submodelpart if no node's ID is provided. The default variable is DISPLACEMENT, and in case no variable is considered for the slave the same variable will be considered",
+            "computing_model_part_name"   : "computing_domain",
+            "model_part_name"             : "please_specify_model_part_name",
+            "new_model_part_name"         : "Rigid_Movement_ModelPart",
```

Comment:
All the process that affect the system of dof require the computing model part, because you pass the model and the model does not know the computing model part (at least now)

Reply:
Ok ... fine. My only concern is that this is extra work with fast transfer between modelparts ... 
When this condition is added to the GiD interface we will have an unique and exclusive submodelpart for this condition so no need and this is extra work. Also when passing a submodelpart to this process, it is evident that process will add constraints (from the comments in the code you made).  
Also in the cases where this submodelpart is used somewhere else, this sub model part is not being modified (in terms of nodes and elements) so  I do not see why it is a problem.

Reply:
What do you suggest?, to use the origin model part?

Reply:
Yes the modelpart with "model_part_name" as name and as you correctly pointed out (which I missed) the computing modelpart !

Reply:
OK

Reply:
Done, I consider the origin model part as destination now by default